Audit detect-secrets scanner results  and update false-positives

simysc opened the pull request:
- [x] I have considered the risk of behavior change or other zero migration impact (https://github.com/OpenLiberty/open-liberty/wiki/Behavior-Changes).
- [x] If this PR fixes an Issue, the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN" (verify `release bug` label if applicable: https://github.com/OpenLiberty/open-liberty/wiki/Open-Liberty-Conventions).
- [x] If this PR resolves an external Known Issue (including APARS), the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN".

################################################################################################

# Delete this section and fill in the remaining info from the template

ATTENTION, READ THIS: Updated July 2024 

Read and understand this completely, then delete the static part of the template. 

If a reviewer or merger sees this template, they should fail the review or merge.

If this code change is fixing a user-visible bug in previously released code, it MUST
have an associated issue **mentioned in the PR text or description**.  
 
   - That Issue also MUST be labelled “release bug”

     This directs automation to scrape this fix for inclusion in the next release's
     list of bugs fixed.

   - The linkage between PR and Issue should use the `Fixes #...` syntax rather than a manual
     link via the Development widget.

Otherwise, a link to an issue or specific issue labels are optional.

For full details, please see this wiki page: https://github.com/OpenLiberty/open-liberty/wiki/Open-Liberty-Conventions
################################################################################################

simysc committed:
Audit detect-secrets scanner results  and update false-positives

mbroz2 commented on .secrets.baseline:
```diff
@@ -27443,10 +27456,9 @@
+    "dev/com.ibm.ws.openapi.ui/swagger-ui/dist/liberty-swagger-ui-bundle-cd2c6b19d75012eb0aa3.js": [
       {
         "hashed_secret": "6a40fbb3bd3c0673c9bff6ad305c8181d648a9bb",
-        "is_secret": false,
```
Was this removed purposefully?  Is it being investigated further?

simysc replied:
Nope..not sure how it was removed. I can add it back.

simysc replied:
I see the file name has been changed in the end (from **bdf9b527ec6da92b71c5.js** to **cd2c6b19d75012eb0aa3.js**).

mbroz2 replied:
It's possible this file name changes periodically (when built or when pulled from upstream).  I suggest checking if @Azquelt to see if that's the case. If it is, and depending on how often this gets updated, it might make sense to add it to the ignore list.

simysc replied:
It makes sense to update the following as a false positive and I did so:
hashed_secret": "6a40fbb3bd3c0673c9bff6ad305c8181d648a9bb

Azquelt replied:
Yes, this is a minified javascript file which includes a hash of the file in the filename to ensure that browsers can cache the file, but won't wrongly use the cache if the file has changed.

Azquelt replied:
Ignoring `dev/com.ibm.ws.openapi.ui/swagger-ui/dist/liberty-swagger-ui-bundle-*.js` should be safe.

simysc committed:
Update .secrets.baseline

Updated "hashed_secret": "6a40fbb3bd3c0673c9bff6ad305c8181d648a9bb" as false positive.

mbroz2 commented on .secrets.baseline:
```diff
@@ -3,7 +3,7 @@
     "files": "package-lock.json|.*\\/OSGI-INF\\/.*\\.properties$|.+\\.(nlsprops)$|^dev\\/(.*(_test|_fat)).*\\/ltpa.keys$|^.secrets.baseline.*|^.secrets.baseline$",
```
```suggestion
    "files": "package-lock.json|.*\\/OSGI-INF\\/.*\\.properties$|.+\\.(nlsprops)$|^dev\\/(.*(_test|_fat)).*\\/ltpa.keys$|^dev\\/com.ibm.ws.openapi.ui\\/swagger-ui\\/dist\\/liberty-swagger-ui-bundle-.*\\.js$|^.secrets.baseline.*|^.secrets.baseline$",
```
The above is hand-crafted, so its correctness would need to be validated.  @simysc would you be able to commit this to your branch and rerun it to check if it correctly ignores that file.

simysc replied:
Done.

mbroz2 replied:
@simysc I don't see a commit added with the above change

simysc replied:
Updated line 3 and verified it worked fine and ignored .js file

simysc committed:
Update .secrets.baseline

I checked and found it ignored the .js file.

mbroz2 reviewed (approved):
Lgtm

simysc commented:
@ayoho can you please review the changes?

simysc commented:
#build ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=1fb338f3-ea42-4f75-a446-52306f4ad66f) - :white_check_mark: completed successfully!)

_Note: Target locations of links might be accessible only to IBM employees._
